### ISSUE TITLE
cumsum of size 1 array + disable_jit returns False

### ISSUE THREAD

**Issue by Joshuaalbert** [google/jax#5497]
``` python
from jax import numpy as jnp, disable_jit

with disable_jit():
  p = jnp.array([1.])
  print(jnp.cumsum(p))
  #False
```
I don't think it should be returning a False.
And if I try to see if it matches False I get this error:
```python
with disable_jit():
  p = jnp.array([1.])
  assert jnp.cumsum(p) == False
```

```bash
2021-01-23 20:42:58.282486: E external/org_tensorflow/tensorflow/compiler/xla/pjrt/pjrt_client.cc:1809] Execution of replica 0 failed: Invalid argument: Argument does not match host shape or layout of computation parameter 0: want f32[1]{0}, got pred[]
Traceback (most recent call last):
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3417, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-47-45e137a16643>", line 3, in <module>
    assert jnp.cumsum(p) == False
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4840, in deferring_binary_op
    return binary_op(self, other)
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 378, in <lambda>
    fn = lambda x1, x2: lax_fn(*_promote_args(numpy_fn.__name__, x1, x2))
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/_src/lax/lax.py", line 372, in eq
    return eq_p.bind(x, y)
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/core.py", line 270, in bind
    out = top_trace.process_primitive(self, tracers, params)
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/core.py", line 580, in process_primitive
    return primitive.impl(*tracers, **params)
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/interpreters/xla.py", line 236, in apply_primitive
    return compiled_fun(*args)
  File "/home/albert/miniconda3/envs/jax_py/lib/python3.8/site-packages/jax/interpreters/xla.py", line 351, in _execute_compiled_primitive
    out_bufs = compiled.execute(input_bufs)
RuntimeError: Invalid argument: Argument does not match host shape or layout of computation parameter 0: want f32[1]{0}, got pred[]
```

**Comment by hawkinsp** [comment]
What jax/jaxlib version and hardware are you using? I can't reproduce this on CPU or GPU.

**Comment by jakevdp** [comment]
I can reproduce on jax v0.2.7, but the issue seems to be fixed already in jax v0.2.8

**Comment by hawkinsp** [comment]
We haven't seen this reoccur and I also cannot reproduce it at head, so I'm going to declare it fixed!

Closing. Please reopen if you can still reproduce this!
